1. Explain the different types of data (qualitative and quantitative) and provide examples of each. Discuss
nominal, ordinal, interval, and ratio scales.

qualitative data>> measurable and contains numerics. it is divided into two types discrete and continuous.

quantitative data>>these are categorical data which contains string. divided into two types nominal and ordinal.

nominal scale data>> these have no order in categorical data eg. blood group, gender.

ordinal data>> these have order in categorical data. eg. Rank, grades.

interval scales>> it doesnt compulsorily have 0 starting value eg. temperature.

ratio scales>> these data must have a zero starting values so that they can be compared. eg.two people weight are 80kg & 40kg.





2. What are the measures of central tendency, and when should you use each? Discuss the mean, median,
and mode with examples and situations where each is appropriate

measures of centeral tendency are mean median mode.

**mean** is calculated to find the average of numbers that is in qualitative data we use it so as to fill the unknown values in the data. but it has a limitations if outliers are present. example age of persons are>> 23,24,22,25,NA,24. now to fill the unkown value we use the mean since the values doesnot contains any outliers.

**median** is calculated to find the middle most value of sorted numeric values. this is generally useful when the outliners are present example weight of people in kg are 40,45,50,48,na,42,120 . since the 120 is an outlier and shifts the mean to higher side. so we calculate man here.

**mode** is the value to most repeated value in the given data. it is generally calculated for the categorical data. example gender in the given data are>> M,M,M,F,na,F,M. So to replace the unknown value mode is calculated and here most repated is male.





3. Explain the concept of dispersion. How do variance and standard deviation measure the spread of data?

dispersion tells how is the data is spread, that is how much the data is deviated from the central value. it tells the variablity in the data. more is the dispersion more is the variation in the data.

**Variance** measures how far each data point is from the mean, on average Variance gives a squared measure of dispersion, meaning its unit is different from the original data.

**Standard Deviation**: The square root of variance.
It represents the average deviation from the mean and has the same unit as the original data, making it more interpretable than variance.

4. What is a box plot, and what can it tell you about the distribution of data?

A box plot (also called a box-and-whisker plot) is a graphical representation of a dataset that displays its spread, central tendency, and potential outliers. It provides a summary of the data using five key statistical measures:

Minimum – The smallest data point (excluding outliers).
First Quartile (Q1) – The 25th percentile (lower quartile).
Median (Q2) – The middle value (50th percentile).
Third Quartile (Q3) – The 75th percentile (upper quartile).
Maximum – The largest data point (excluding outliers).

**Symmetry & Skewness**:
If the median is centered in the box, the distribution is symmetric.
If the median is closer to Q1 or Q3, the data is skewed.

**Outliers**: Dots outside the whiskers indicate outliers, which are unusually high or low values


5. Discuss the role of random sampling in making inferences about populations.

Every member has equal probablity of getting selected in the sample

Random sampling helps eliminate selection bias, ensuring that the sample represents the overall population rather than specific subgroups.
This improves the validity of conclusions drawn from the sample.
for example Opinion polls use random sampling to estimate public opinion.

6. Explain the concept of skewness and its types. How does skewness affect the interpretation of data?

The skewness in data determines the extent of symmetricity around its mean. if there is no skewness is the data means that the data is symmetric about mean and has mean=median=mode it is of following types

Positive Skew (Right-Skewed): The tail is longer on the right; mean > median > mode (e.g., income distribution).

Negative Skew (Left-Skewed): The tail is longer on the left; mean < median < mode (e.g., exam scores where many students score high).

Zero Skew (Symmetric): The data is evenly distributed around the mean (e.g., normal distribution).

Some of the machine learning algorithms requires symmetric data to build the the ml models if not symmetric so the transformation in the data is done.


7. What is the interquartile range (IQR), and how is it used to detect outliers?

Itis the difference of 3rd quatile(75%ile) and 2nd quatile(25%ile).

Any data point outside the range [Q1 - 1.5 × IQR, Q3 + 1.5 × IQR] is considered an outlier.




8. Discuss the conditions under which the binomial distribution is used.
Fixed number of trials (n) – A predetermined number of independent experiments.
Two possible outcomes (Success/Failure) – Each trial results in success (p) or failure (1-p).
Constant probability of success (p) – The probability remains the same across all trials.



9. Explain the properties of the normal distribution and the empirical rule (68-95-99.7 rule).

Properties:

Bell-shaped and symmetric around the mean.
Mean = Median = Mode.
Defined by mean (μ) and standard deviation (σ).
The total area under the curve is 1.

Empirical Rule (68-95-99.7 Rule):
68% of data lies within 1σ of the mean.
95% lies within 2σ.
99.7% lies within 3σ.
Helps in probability estimation for normal distributions.


10. Provide a real-life example of a Poisson process and calculate the probability for a specific event.

A Poisson process models the occurrence of rare events in a fixed time or space, assuming:

Events are independent.

The rate (λ) is constant.

Example: A call center receives an average of 5 calls per hour (λ = 5).

What’s the probability of receiving exactly 3 calls in an hour?

Interpretation: There is a 14.04% chance of receiving exactly 3 calls in an hour.

11. Explain what a random variable is and differentiate between discrete and continuous random variables.

A random variable assigns numerical values to outcomes in a probabilistic experiment.

Discrete random variable:
Takes countable values (e.g., number of heads in coin flips).
Example: X = {0,1,2,3,4} (where X is the number of defective products in a batch).

Continuous Random Variable:
Takes infinite values in a range (e.g., height, temperature).
Example: Y = height of a randomly selected person (Y ∈ [140 cm, 200 cm]).

In [8]:
#12. Provide an example dataset, calculate both covariance and correlation, and interpret the results.

import numpy as np
import pandas as pd
X = np.array([1, 2, 3, 4, 5])
Y = np.array([2, 3, 6, 8, 10])

cov_matrix = np.cov(X, Y, bias=False)  # bias=False gives sample covariance
covariance = cov_matrix[0, 1]

correlation = np.corrcoef(X, Y)[0, 1]

print(f"Covariance: {covariance}")
print(f"Correlation: {correlation}")

Covariance: 5.25
Correlation: 0.9921567416492214


Covariance (5.25) indicates a positive relationship but lacks scale interpretation.

Correlation (0.97) suggests a strong positive linear relationship between X and Y.